Apps/phcuckoo/zip_and_encrypt.py

```diff
@@ -0,0 +1,111 @@
+# File: zip_and_encrypt.py
+#
+# Copyright (c) 2014-2021 Splunk Inc.
+#
+# Licensed under Apache 2.0 (https://www.apache.org/licenses/LICENSE-2.0.txt)
+#
+
+import sys
+import os
+import uuid
+import subprocess
+
+zipexec = "/usr/bin/zip"
+
+
+def _is_exec(path):
+    if not os.path.isfile(path) or not os.access(path, os.X_OK):
+        return False
+    return True
+
+
+class zip_and_encrypt:
+    def __init__(self, prefix, password):
+        if not _is_exec(zipexec):
+            raise Exception(f"{zipexec} is not executable")
+
+        self._prefix = prefix
+        self._password = password
+        self._archive = f"{prefix}_{uuid.uuid4()}"
+        self._fp = None
+        self._output = None
+
+    @property
+    def output(self):
+        return self._output
+
+    @property
+    def archive_name(self):
+        return self._archive
+
+    @property
+    def archive_path(self):
+        return self._archive + ".zip"
+
+    @property
+    def archive_fp(self):
+        if self._fp:
+            return self._fp
+
+        if not os.path.isfile(self.archive_path):
+            raise Exception(f"{self.archive_path} doesn't exists")
+
+        self._fp = open(self.archive_path, "rb")
+        return self._fp
+
+    def add(self, filename):
+        if not os.path.exists(filename):
+            raise Exception(f"{filename} does not exists")
+
+        cmd = [
+            zipexec,
+            "-j",
+            "-P",
+            self._password,
```

Comment:
(copying comment from #335 since the app got moved to this PR) 

can we avoid passing a password in the command and pass it through stdin instead?

looking at zip's man page (https://linux.die.net/man/1/zip) I think using the -e flag should achieve this. although you may need to use something like pexpect to handle the interaction if direct piping doesn't work

Reply:
Let us look into the suggestion you have provided, we have checked -e flag but we have to figure out that how can we handle prompting for a password without user-interaction while the action is running. Because using this flag, the command will display a prompt to enter a password first, but in our case what should be the case that we have to decide.

Reply:
As per your suggestion, we have updated command by using -e flag and handled the interaction with pexpect.run() from  pexpect library. Could you please review it? If any changes, please suggest.